source/aws_signing.c

```diff
@@ -1707,6 +1707,123 @@ static int s_build_canonical_request_hash(struct aws_signing_state_aws *state) {
     return result;
 }
 
+/**
+ * Note that there is no canonical request for event signing.
+ * The string to sign for events is detailed here:
+ * https://docs.aws.amazon.com/transcribe/latest/dg/streaming-http2.html
+ *
+ *      String stringToSign =
+ *      "AWS4-HMAC-SHA256" +
+ *      "\n" +
+ *      DateTime +
+ *      "\n" +
+ *      Keypath +
+ *      "\n" +
+ *      Hex(priorSignature) +
+ *      "\n" +
+ *      HexHash(nonSignatureHeaders) +
+ *      "\n" +
+ *      HexHash(payload);
+ *
+ * This function will build the string_to_sign_payload,
+ * aka "everything after the Keypath line in the string to sign".
+ */
+static int s_build_string_to_sign_payload_for_event(struct aws_signing_state_aws *state) {
+    int result = AWS_OP_ERR;
+
+    struct aws_byte_buf *dest = &state->string_to_sign_payload;
+
+    /*
+     * Hex(priorSignature) + "\n"
+     *
+     * Fortunately, the prior signature is already hex.
+     */
+    struct aws_byte_cursor prev_signature_cursor;
+    AWS_ZERO_STRUCT(prev_signature_cursor);
+    if (aws_signable_get_property(state->signable, g_aws_previous_signature_property_name, &prev_signature_cursor)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_SIGNING, "(id=%p) Event signable missing previous signature property", (void *)state->signable);
+        return aws_raise_error(AWS_AUTH_SIGNING_MISSING_PREVIOUS_SIGNATURE);
+    }
+
+    /* strip any padding (AWS_SIGV4A_SIGNATURE_PADDING_BYTE) from the previous signature */
+    prev_signature_cursor = aws_trim_padded_sigv4a_signature(prev_signature_cursor);
+
+    if (aws_byte_buf_append_dynamic(dest, &prev_signature_cursor)) {
+        return AWS_OP_ERR;
+    }
+
+    if (aws_byte_buf_append_byte_dynamic(dest, '\n')) {
+        return AWS_OP_ERR;
+    }
+
+    /*
```

Comment:
Would it be worth breaking this out into a separate encode style function?

Is there any relationship between these "events" and what we have in aws-c-event-stream?

Reply:
The logic of encoding date header is same as in aws-c-event-stream. I think taking a dependency on aws-c-event-stream is not worth it for a few lines encoding.

Reply:
Gonna disagree with that, given how the CRTs are constructed.

Reply:
The monolith nature of the CRT is a recurring pain point. I think we should not make the problem worse. We can reconsider this choice if this becomes more involved than encoding a single date header.

Reply:
aws-c-event-stream is already in every CRT with the exception of JS, where it will be in soon.

Reply:
Discussed offline that we will not be taking dependency on event-stream for now.